### PR TITLE
Change json lib

### PR DESCRIPTION
**Pull request by zhiltsov-max**
### Summary

For original COCO dataset reading (350 MB json) the speed-up is ~30%, for writing - 10x (which gives ~2.5x speed-up in export). There are some features that can be useful, like builtin `numpy` support.
https://github.com/ijl/orjson

- Replace standard `json` library with `orjson`
- Updated most places (except OpenImages format, where it's not a drop-in replacement)

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: datumaro/cli/commands/detect_format.py [hunk 1]
```diff
@@ -3,7 +3,8 @@
 # SPDX-License-Identifier: MIT
 
 import argparse
-import json
+
+import orjson
```
Comment (nmanovic) [inline datumaro/cli/commands/detect_format.py:7]: Why not `import orjson as json`?

Comment (nmanovic) [reply datumaro/cli/commands/detect_format.py:7]: OK. I see  that it has slightly different interface.

File: datumaro/cli/commands/diff.py [hunk 2]
```diff
@@ -202,8 +203,9 @@ def diff_command(args):
         output_file = osp.join(dst_dir,
             generate_next_file_name('diff', ext='.json', basedir=dst_dir))
         log.info("Saving diff to '%s'" % output_file)
-        with open(output_file, 'w') as f:
-            json.dump(output, f, indent=4, sort_keys=True)
+        with open(output_file, 'wb') as f:
+            f.write(orjson.dumps(output,
+                option=orjson.OPT_SORT_KEYS | orjson.OPT_INDENT_2))
```
Comment (IRDonch) [inline datumaro/cli/commands/diff.py:208]: IMO, we need to add utility functions for common operations (i.e. "save JSON to file" and "load JSON from file") and replace all these chunks of code with calls to those functions. That way:

1. We eliminate repetitive code.
2. We'll be able to easily swap out the JSON implementation later if we need to.

Comment (zhiltsov-max) [reply datumaro/cli/commands/diff.py]: Done

**Pull request merged**